docs/application/dotnet/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,180 @@
+    ```
+
+2. You can construct a pipeline with the description string, including the neural network framework and a model file:
+
+    ```C#
+    /* Create Pipeline instance with the pipeline description. */
+    string description = "appsrc name=srcx ! " +
+                         "other/tensor,dimension=(string)3:224:224:1,type=(string)uint8,framerate=(fraction)0/1 ! " +
+                         "tensor_filter framework=tensorflow-lite model=" + model_path + " ! " +
+                         "tensor_sink name=sinkx";
+    Pipeline pipe = new Pipeline(description);
+    ```
+
+<a name="inputdata"></a>
+## Pushing the input data
+
+`SourceNode` provides a method to push input data into the pipeline:
+
+```C#
+/* Get the source node with the name 'srcx'. */
+var src_node = pipe.GetSource("srcx");
+
+/* Start the pipeline. */
+pipe.Start();
+
+/* Prepare input tensor data, with data type UInt8 and dimension 3:224:224:1. */
+var in_info = new TensorsInfo();
+in_info.AddTensorInfo(TensorType.UInt8, new int[4] { 3, 224, 224, 1 });
+
+var in_data = in_info.GetTensorsData();
+
+/* Push input tensor data into the pipeline. */
+src_node.Input(in_data);
+```
+
+<a name="outputdata"></a>
+## Subscribing to the output data
+
+You can use `SinkNode` to get the output tensor data from the pipeline:
+
+```C#
+/* Firstly, declare the event handler for the data received event. */
+void DataReceivedEvent(object sender, DataReceivedEventArgs args)
+{
+    /* Add your task with received data. */
+    TensorsData dataReceived = args.Data;
+}
+```
+
+```C#
+/* Get the sink node with the name 'sinkx'. */
+var sink_node = pipe.GetSink("sinkx");
+
+/* Subscribe the event to get the data from the sink node. */
+sink_node.DataReceived += DataReceivedEvent;
+
+/* Start the pipeline. */
+pipe.Start();
+```
+
+<a name="dataflow"></a>
+## Data flow control
+
+You need to stop the data stream or select the data flow with multiple stream paths for neural network models. Pipeline provides methods to handle the flow.
+
+1. Start or stop the data flow.
+
+    `Start()` and `Stop()` control the overall data flow of the pipeline asynchronously.
+    You can also get the pipeline state from the `State` property:
+
+    ```C#
+    string description = "input-selector name=ins ! tensor_converter ! valve name=valvex ! tensor_sink name=sinkx " +
+                         "videotestsrc is-live=true ! videoconvert ! ins.sink_0 " +
+                         "videotestsrc is-live=true ! videoconvert ! ins.sink_1";
+    var pipe = new Pipeline(description);
+
+    /* After constructing the pipeline, the pipeline state is 'Paused'. */
+    var pipeline_state = pipe.State;
+
+    /* Start the pipeline (The pipeline state will be 'Playing'). */
+    pipe.Start();
+
+    /* Stop the pipeline (The pipeline state will be 'Paused'). */
+    pipe.Stop();
+    ```
+
+2. Open or close the stream.
+
+    The 'valve' element controls the stream of a pipeline.
```

Comment:
Thank you for updating the sentence.
Please add "the" before "pipeline description".